qiskit/result/utils.py

```diff
@@ -56,6 +62,15 @@ def marginal_counts(result, indices=None, inplace=False, format_marginal=False):
             experiment_result.header.memory_slots = len(indices)
             csize = experiment_result.header.creg_sizes
             experiment_result.header.creg_sizes = _adjust_creg_sizes(csize, indices)
+
+            if getattr(experiment_result.data, "memory", None) is not None:
+                sorted_indices = sorted(indices, reverse=True)  # same convention as for the counts
```

Comment:
`indices` defaults to `None`, what happens here if a user doesn't have indices set?

Reply:
If fails (but it also fails in the current main branch). I will fix this and and a test.